viewing_party/party.py

```diff
@@ -0,0 +1,253 @@
+        return None
+
+    if title:
+        movie["title"] = title
+
+    if genre:
+        movie["genre"] = genre
+
+    if rating:
+        movie["rating"] = rating
+
+    return movie
+
+
+# 2
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+
+    return user_data
+
+
+# 3
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+# 4
+
+def watch_movie(user_data, title):
+    watched = user_data["watched"]
+    watch_list = user_data["watchlist"]
+
+    for index, movie_info in enumerate(watch_list):
+        if title == movie_info["title"]:
+            watched.append(movie_info)
+            user_data["watched"] = watched
+            del watch_list[index]
+    return user_data
+
+
+# Wave 2 **********************************************
+
+# 1
+
+def get_watched_avg_rating(user_data):
+    watched = user_data["watched"]
+    ratings = []
+    for movie_info in (watched):
+        ratings.append(movie_info["rating"])
+
+    if len(watched) == 0:
+        return 0.0
+    else:
+        avg_rating = sum(ratings)/len(ratings)
+        return avg_rating
+
+
+# 2
+
+def get_most_watched_genre(user_data):
+    watched = user_data["watched"]
+    genres = []
+    for movie_info in watched:
+        genres.append(movie_info["genre"])
+
+    fav_genres = {}
+    if len(watched) == 0:
+        return None 
+    else:
+        for genre in genres:
+            if genre not in fav_genres:
+                fav_genres[genre] = 1
+            else:
+                fav_genres[genre] += 1
+    
+    favorite_genre = max(fav_genres, key=fav_genres.get)
+    
+    return favorite_genre
+
+
+# Wave 3 **********************************************
+
+# 1
+
+def get_unique_watched(user_data):
+    unique_watched_list = []
+    for dicts in user_data["watched"]:
+        unique_watched_list.append(dicts)
+    
+    friends_watched_list = []
+    for friends_watched in user_data["friends"]:
+        for friends_dicts in friends_watched["watched"]:
+            friends_watched_list.append(friends_dicts)
+
+    unique_watched_set = set()
+    friends_watched_set = set()
+
+    if len(user_data["watched"]) == 0:
+        return []
+    else:
+        for movie_dict in unique_watched_list:
+            for movie in movie_dict.values():
+                unique_watched_set.add(movie)
+        for friends_movie_dicts in friends_watched_list:
+            for title in friends_movie_dicts.values():
+                friends_watched_set.add(title)
+
+    set_difference = unique_watched_set - friends_watched_set
+    unique_watched_list = list(set_difference)
+
+    users_unique_watched_list = []
+    for unique_movies in unique_watched_list:
+        list_of_dicts = {"title": unique_movies}
+        users_unique_watched_list.append(list_of_dicts)
+
+    return users_unique_watched_list
+
+
+# 2
+
+def get_friends_unique_watched(user_data):
```

Comment:
Similar feedback for the previous function applies to this function as well. We'd like to return the actual dictionaries from the user_data structure, which might have additional data keys in "real" data.